internal/provider/resource_fmc_network_groups.go

```diff
@@ -0,0 +1,675 @@
+	if diags.HasError() {
+		return state, diags
+	}
+
+	for _, bulk := range bulks {
+		readable := slices.Clone(bulk.groups)
+		for i := range readable {
+			readable[i].json = ""
+		}
+		tflog.Debug(ctx, fmt.Sprintf("%s: bulk ordered for Create: %+v", plan.Id.ValueString(), readable))
+	}
+
+	for _, bulk := range bulks {
+		state, diags = bulk.Create(ctx, plan, state, r.client, reqMods...)
+		if diags.HasError() {
+			return state, diags
+		}
+	}
+
+	// Subresources to Update.
+	tflog.Debug(ctx, fmt.Sprintf("%s: considering remaining subresources for Update: %+v", plan.Id.ValueString(), seq))
+	for _, group := range seq {
+		ok, diags := isConfigUpdatingAt(ctx, tfsdkPlan, tfsdkState, path.Root("items").AtMapKey(group.name))
+		if diags.HasError() {
+			return state, diags
+		}
+
+		if !ok {
+			continue
+		}
+
+		updating := plan.Items[group.name].Id.ValueString()
+		tflog.Debug(ctx, fmt.Sprintf("%s: Subresource %s: Beginning Update", updating, group.name))
+
+		body, diags := group.Body(ctx, plan)
+		if diags.HasError() {
+			return state, diags
+		}
+
+		res, err := r.client.Put(plan.getPath()+"/"+url.QueryEscape(updating), body, reqMods...)
+		if err != nil {
+			return state, diag.Diagnostics{
+				diag.NewErrorDiagnostic("Client Error", fmt.Sprintf("Failed to configure object (PUT), got error: %s, %s", err, res.String())),
+			}
+		}
+
+		state.Items[group.name] = plan.Items[group.name]
+
+		tflog.Debug(ctx, fmt.Sprintf("%s: Subresource %s: Update finished successfully", updating, group.name))
+	}
+
+	// Subresources to Delete.
+	stateBody := state.toBody(ctx, NetworkGroups{})
+	delSeq, diags := graphTopologicalSeq(ctx, stateBody)
+	if diags.HasError() {
+		return state, diags
+	}
+
+	for i := len(delSeq) - 1; i >= 0; i-- {
+		gn := delSeq[i].name
+		if _, found := plan.Items[gn]; found {
+			// item present both in state and in plan, do not delete
+			continue
+		}
+
+		deleting := state.Items[gn].Id.ValueString()
+		tflog.Debug(ctx, fmt.Sprintf("%s: Subresource %s: Beginning Delete", deleting, gn))
+
+		res, err := r.client.Delete(state.getPath()+"/"+url.QueryEscape(deleting), reqMods...)
+		if err != nil {
+			return state, diag.Diagnostics{
+				diag.NewErrorDiagnostic("Client Error", fmt.Sprintf("Failed to delete object (DELETE), got error: %s, %s", err, res.String())),
+			}
+		}
+
+		delete(state.Items, gn)
+		tflog.Debug(ctx, fmt.Sprintf("%s: Subresource %s: Delete finished successfully", deleting, gn))
+	}
+
+	return state, nil
+}
+
+func isConfigUpdatingAt(ctx context.Context, tfsdkPlan tfsdk.Plan, tfsdkState tfsdk.State, where path.Path) (bool, diag.Diagnostics) {
+	var pv, sv attr.Value
+
+	diags := tfsdkPlan.GetAttribute(ctx, where, &pv)
+	if diags.HasError() {
+		return false, diags
+	}
+
+	diags = tfsdkState.GetAttribute(ctx, where, &sv)
+	if diags.HasError() {
+		return false, nil
+	}
+
+	return !sv.Equal(pv), diags
+}
+
+// group is an internal representation of a single fmc_network_group.
+type group struct {
```

Comment:
As all the code in `internal/provider` is in the same go package, we should be careful to not create naming collisions. So we should for example prefix this name with the resource type.

Reply:
Done, thanks.